libraries/surplus/src/basic-tests.js

```diff
@@ -0,0 +1,179 @@
+  ComponentWithChildren,
+  ComponentWithChildrenRerender,
+  ComponentWithDifferentViews,
+  ComponentWithProperties,
+  ComponentWithUnregistered,
+  ComponentWithImperativeEvent,
+  ComponentWithDeclarativeEvent
+} from "./components";
+
+//let _it = it;
+//it = (title, fn) => _it(title, () => S.root(fn));
+
+describe("basic support", function() {
+
+  describe("no children", function() {
+    it("can display a Custom Element with no children", function() {
+      S.root(() => {
+        this.weight = 3;
+        let root = <ComponentWithoutChildren />;
+        let wc = root.wc;
+        expect(wc).toExist();
+      });
+    });
+  });
+
+  describe("with children", function() {
+    function expectHasChildren(wc) {
+      expect(wc).toExist();
+      let shadowRoot = wc.shadowRoot;
+      let heading = shadowRoot.querySelector("h1");
+      expect(heading).toExist();
+      expect(heading.textContent).toEqual("Test h1");
+      let paragraph = shadowRoot.querySelector("p");
+      expect(paragraph).toExist();
+      expect(paragraph.textContent).toEqual("Test p");
+    }
+
+    it("can display a Custom Element with children in a Shadow Root", function() {
+      S.root(() => {
+        this.weight = 3;
+        let root = <ComponentWithChildren />;
+        let wc = root.wc;
+        expectHasChildren(wc);
+      });
+    });
+
+    it("can display a Custom Element with children in a Shadow Root and pass in Light DOM children", async function() {
+      S.root(async () => {
+        this.weight = 3;
+        let root = <ComponentWithChildrenRerender />;
+        let wc = root.wc;
+        await Promise.resolve();
+        expectHasChildren(wc);
+        expect(wc.textContent.includes("2")).toEqual(true);
+      });
+    });
+
+    it("can display a Custom Element with children in the Shadow DOM and handle hiding and showing the element", function() {
+      S.root(() => {
+        this.weight = 3;
+        let { root, toggle } = <ComponentWithDifferentViews />;
+        let wc = root.wc;
+        expectHasChildren(wc);
+        toggle();
+        let dummy = root.dummy;
+        expect(dummy).toExist();
+        expect(dummy.textContent).toEqual("Dummy view");
+        toggle();
+        wc = root.wc;
+        expectHasChildren(wc);
+      });
+    });
+  });
+
+  describe("attributes and properties", function() {
+    it("will pass boolean data as either an attribute or a property", function() {
+      S.root(() => {
+        this.weight = 3;
+        let root = <ComponentWithProperties />;
+        let wc = root.wc;
+        let data = wc.bool || wc.hasAttribute("bool");
+        expect(data).toBe(true);
+      });
+    });
+
+    it("will pass numeric data as either an attribute or a property", function() {
+      S.root(() => {
+        this.weight = 3;
+        let root = <ComponentWithProperties />;
+        let wc = root.wc;
+        let data = wc.num || wc.getAttribute("num");
+        expect(data).toEqual(42);
```

Comment:
I know it's super annoying, but recently we switched from expect.js to chai for expectations.

For example, you can see how we're handling a similar test case in react now: https://github.com/webcomponents/custom-elements-everywhere/blob/master/libraries/react/src/basic-tests.js#L118

I think the PR will need to be updated to use chai.

Reply:
Sure, can do.  Is https://github.com/webcomponents/custom-elements-everywhere/commit/e294e8d150ba02ea57f6e5119056e5f09f92d9e5#diff-aca65ac5cb69fb4b776030c3d7659d70 a good reference for the changes?